editor/src/clj/editor/collection_non_editable.clj

```diff
@@ -211,24 +209,27 @@
     (distinct)
     instance-property-descs))
 
-(defn- collection-desc->referenced-collection-proj-paths [collection-desc]
-  (eduction
-    (map :collection)
-    (distinct)
-    (:collection-instances collection-desc)))
+(defn- collection-desc->referenced-collection-proj-path->index [collection-desc]
+  (into {}
+        (comp (map :collection)
+              (distinct)
+              (map-indexed flipped-pair))
+        (:collection-instances collection-desc)))
 
-(defn- collection-desc->referenced-game-object-proj-paths [collection-desc]
-  (eduction
-    (map :prototype)
-    (distinct)
-    (:instances collection-desc)))
+(defn- collection-desc->referenced-game-object-proj-path->index [collection-desc]
+  (into {}
+        (comp (map :prototype)
+              (distinct)
+              (map-indexed flipped-pair))
+        (:instances collection-desc)))
```

Comment:
The same applies to any referenced game objects. We need to be able to resolve the reference to a particular source game object resource in order to produce our scene.